### PR TITLE
Security flaws check improvements

### LINKED ISSUE #4283: Improve the `test_check_security_flaws` code scan
## Description

We detect that some enhancements can be made to the `test_check_security_flaws` function to improve the test execution.

#### 1. Fix the values of `DEFAULT_DIRECTORIES_TO_EXCLUDE`

https://github.com/wazuh/wazuh-qa/blob/cbd788cb181a92be6020673e91c3a2c06d7487b9/tests/scans/code_analysis/conftest.py#L7

These default values are giving problems in some cases and aren't being taken into account. Making the test fails, because analyzes files that should not be watching.

```
pytest tests/scans/code_analysis/test_python_flaws.py --disable-warnings
```

<details><summary>Test Output</summary>
<p>

```python
============================= test session starts ==============================
platform linux -- Python 3.9.9, pytest-6.2.2, py-1.10.0, pluggy-0.13.1
rootdir: /home/nstefani/git/wazuh-qa/tests, configfile: pytest.ini
plugins: html-3.1.1, metadata-2.0.2, testinfra-5.0.0
collected 1 item

tests/scans/code_analysis/test_python_flaws.py F                         [100%]

=================================== FAILURES ===================================
__________________________ test_check_security_flaws ___________________________

clone_wazuh_repository = '/tmp/tmpt8f5frne'
get_test_parameters = {'directories_to_check': ['framework/', 'api/', 'wodles/'], 'directories_to_exclude': 'tests/,test/', 'min_confidence_level': 'MEDIUM', 'min_severity_level': 'LOW', ...}

    def test_check_security_flaws(clone_wazuh_repository, get_test_parameters):
        """Test whether the directory to check has python files with possible vulnerabilities or not.
    
        The test passes if there are no new vulnerabilities. The test fails in other case and generates a report.
    
        In case there is at least one vulnerability, a json file will be generated with the report. If we consider this
        result or results are false positives, we will move the json object containing each specific result to the
        `known_flaws/known_flaws_{framework|api|wodles}.json` file.
    
        Args:
            clone_wazuh_repository (fixture): Pytest fixture returning the path of the temporary directory path the
                repository cloned. This directory is removed at the end of the pytest session.
            get_test_parameters (fixture): Pytest fixture returning the a dictionary with all the test parameters.
                These parameters are the directories to check, directories to exclude, the minimum confidence level, the
                minimum severity level and the repository name.
        """
        # Wazuh is cloned from GitHub using the clone_wazuh_repository fixture
        assert clone_wazuh_repository, "Error while cloning the Wazuh repository from GitHub, " \
                                       "please check the Wazuh branch set in the parameter."
        # Change to the cloned Wazuh repository directory
        os.chdir(clone_wazuh_repository)
    
        directories_to_check = get_test_parameters['directories_to_check']
        bandit_output_list = \
            run_bandit_multiple_directories(directories_to_check,
                                            get_test_parameters['directories_to_exclude'],
                                            get_test_parameters['min_severity_level'],
                                            get_test_parameters['min_confidence_level'])
    
        flaws_already_found = {}
        for bandit_output, directory in zip(bandit_output_list, directories_to_check):
            assert not bandit_output['errors'], \
                f"\nBandit returned errors when trying to get possible vulnerabilities in the directory " \
                f"{directory}:\n{bandit_output['errors']}"
    
            bandit_result = bandit_output['results']
    
            known_flaws = update_known_flaws_in_file(known_flaws_directory=KNOWN_FLAWS_DIRECTORY,
                                                     directory=directory,
                                                     is_default_check_dir=
                                                     directory.replace('/', '') in
                                                     DEFAULT_DIRECTORIES_TO_CHECK.replace('/', '').split(','),
                                                     bandit_results=bandit_result)
    
            flaws_already_found = get_new_flaws(bandit_results=bandit_result,
                                                known_flaws=known_flaws,
                                                directory=directory,
                                                flaws_already_found=flaws_already_found,
                                                new_flaws_output_dir=TEST_PYTHON_CODE_PATH)
    
>       assert not any(
            flaws_already_found.get(directory, None) for directory in directories_to_check), \
            f"\nThe following possible vulnerabilities were found: {json.dumps(flaws_already_found, indent=4, sort_keys=True)}"
E       AssertionError: 
E         The following possible vulnerabilities were found: {
E             "framework/": "Vulnerabilities found in files: framework/scripts/tests/test_agent_upgrade.py, framework/scripts/tests/test_cluster_control.py, framework/scripts/tests/test_rbac_control.py, framework/scripts/tests/test_wazuh_clusterd.py, framework/scripts/tests/test_wazuh_logtest.py, framework/wazuh/core/cluster/dapi/tests/test_dapi.py, framework/wazuh/core/cluster/tests/test_client.py, framework/wazuh/core/cluster/tests/test_cluster.py, framework/wazuh/core/cluster/tests/test_common.py, framework/wazuh/core/cluster/tests/test_control.py, framework/wazuh/core/cluster/tests/test_local_client.py, framework/wazuh/core/cluster/tests/test_local_server.py, framework/wazuh/core/cluster/tests/test_master.py, framework/wazuh/core/cluster/tests/test_server.py, framework/wazuh/core/cluster/tests/test_utils.py, framework/wazuh/core/cluster/tests/test_worker.py, framework/wazuh/core/tests/test_active_response.py, framework/wazuh/core/tests/test_agent.py, framework/wazuh/core/tests/test_cdb_list.py, framework/wazuh/core/tests/test_common.py, framework/wazuh/core/tests/test_configuration.py, framework/wazuh/core/tests/test_database.py, framework/wazuh/core/tests/test_decoder.py, framework/wazuh/core/tests/test_exception.py, framework/wazuh/core/tests/test_logtest.py, framework/wazuh/core/tests/test_manager.py, framework/wazuh/core/tests/test_mitre.py, framework/wazuh/core/tests/test_pyDaemonModule.py, framework/wazuh/core/tests/test_results.py, framework/wazuh/core/tests/test_rootcheck.py, framework/wazuh/core/tests/test_rule.py, framework/wazuh/core/tests/test_sca.py, framework/wazuh/core/tests/test_security.py, framework/wazuh/core/tests/test_stats.py, framework/wazuh/core/tests/test_syscheck.py, framework/wazuh/core/tests/test_syscollector.py, framework/wazuh/core/tests/test_task.py, framework/wazuh/core/tests/test_utils.py, framework/wazuh/core/tests/test_vulnerability.py, framework/wazuh/core/tests/test_wazuh_queue.py, framework/wazuh/core/tests/test_wazuh_socket.py, framework/wazuh/core/tests/test_wdb.py, framework/wazuh/core/tests/test_wlogging.py, framework/wazuh/rbac/tests/test_auth_context.py, framework/wazuh/rbac/tests/test_decorators.py, framework/wazuh/rbac/tests/test_default_configuration.py, framework/wazuh/rbac/tests/test_orm.py, framework/wazuh/rbac/tests/test_preprocessor.py, framework/wazuh/tests/test_active_response.py, framework/wazuh/tests/test_agent.py, framework/wazuh/tests/test_cdb_list.py, framework/wazuh/tests/test_ciscat.py, framework/wazuh/tests/test_cluster.py, framework/wazuh/tests/test_decoder.py, framework/wazuh/tests/test_group.py, framework/wazuh/tests/test_logtest.py, framework/wazuh/tests/test_manager.py, framework/wazuh/tests/test_mitre.py, framework/wazuh/tests/test_rootcheck.py, framework/wazuh/tests/test_rule.py, framework/wazuh/tests/test_sca.py, framework/wazuh/tests/test_security.py, framework/wazuh/tests/test_stats.py, framework/wazuh/tests/test_syscheck.py, framework/wazuh/tests/test_syscollector.py, framework/wazuh/tests/test_task.py, framework/wazuh/tests/test_vulnerability.py, check them in /home/nstefani/git/wazuh-qa/tests/scans/code_analysis/new_flaws_framework.json",
E             "wodles/": "Vulnerabilities found in files: wodles/aws/tests/test_aws.py, wodles/azure/tests/test_azure.py, wodles/azure/tests/test_orm.py, wodles/docker-listener/tests/test_docker_listener.py, wodles/gcloud/tests/test_bucket.py, wodles/gcloud/tests/test_gcloud.py, wodles/gcloud/tests/test_integration.py, wodles/gcloud/tests/test_subscriber.py, wodles/gcloud/tests/test_tools.py, check them in /home/nstefani/git/wazuh-qa/tests/scans/code_analysis/new_flaws_wodles.json"
E         }
E       assert not True
E        +  where True = any(<generator object test_check_security_flaws.<locals>.<genexpr> at 0x7fc85b2d1c80>)

/home/nstefani/git/wazuh-qa/tests/scans/code_analysis/test_python_flaws.py:64: AssertionError
=========================== short test summary info ============================
FAILED tests/scans/code_analysis/test_python_flaws.py::test_check_security_flaws
======================== 1 failed, 3 warnings in 36.81s ========================

```

</p>
</details> 

#### 2. Remove warning from test execution

```python
(wqa) ➜  wazuh-qa git:(master) ✗ date; pytest tests/scans/code_analysis/test_python_flaws.py --exclude_directories tests,test
mié 05 jul 2023 17:06:46 -03
========================================== test session starts ===========================================
platform linux -- Python 3.9.9, pytest-6.2.2, py-1.10.0, pluggy-0.13.1
rootdir: /home/nstefani/git/wazuh-qa/tests, configfile: pytest.ini
plugins: html-3.1.1, metadata-2.0.2, testinfra-5.0.0
collected 1 item

tests/scans/code_analysis/test_python_flaws.py .                                                   [100%]

============================================ warnings summary ============================================
scans/code_analysis/test_python_flaws.py::test_check_security_flaws
  invalid escape sequence \w

scans/code_analysis/test_python_flaws.py::test_check_security_flaws
  invalid escape sequence \g

scans/code_analysis/test_python_flaws.py::test_check_security_flaws
  invalid escape sequence \d

-- Docs: https://docs.pytest.org/en/stable/warnings.html
===================================== 1 passed, 3 warnings in 12.31s =====================================
```

Currently, we have three warnings, from the test execution, that easily we can fix.

### Expected behavior

1. The tests should skip the specified directories, replacing the default values with `tests,test`.
2.  The execution of the test should not show any warning.

### PR DESCRIPTION
**Pull request by nico-stefani**
|Related issue|
|-------------|
|   #4283  |

## Description

This PR closes #4283. Adds some fixes and improvements for the `test_python_flaws.py`.

I was not able to find the reason for the warnings because these are not present in the `tests_python_flaws.py` and neither is present the line number of the warning in the report. 
So I decided that we can continue using the `--disable-warnings` option since this is not a major issue.

### Updated

- Update readme replacing `branch` with `reference`.
- Fix the default value for the `reference` option.
- Fix the format value of directories to exclude default.

---

## Testing performed

```
(wqa310) ➜  wazuh-qa git:(4283-fix-security-flaws-check) ✗ pytest tests/scans/code_analysis/test_python_flaws.py
================================================================================================ test session starts ================================================================================================
platform linux -- Python 3.10.6, pytest-7.1.2, pluggy-1.2.0
rootdir: /home/nstefani/git/wazuh-qa/tests, configfile: pytest.ini
plugins: html-3.1.1, metadata-3.0.0, testinfra-5.0.0
collected 1 item

tests/scans/code_analysis/test_python_flaws.py .                                                                                                                                                              [100%]

================================================================================================= warnings summary ==================================================================================================
scans/code_analysis/test_python_flaws.py::test_check_security_flaws
  invalid escape sequence '\w'

scans/code_analysis/test_python_flaws.py::test_check_security_flaws
  invalid escape sequence '\g'

scans/code_analysis/test_python_flaws.py::test_check_security_flaws
  invalid escape sequence '\d'

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================================================================================== 1 passed, 3 warnings in 8.69s ===========================================================================================
```

### CODE REVIEW

**Review by fdalmaup — DISMISSED** [review]
LGTM!

File: tests/scans/code_analysis/README.md [hunk 1]
```diff
@@ -32,29 +32,29 @@ The workaround for this test will be the following:
...
   - We analyze the new vulnerabilities found in the module and report them in GitHub issues.
   - We move the vulnerabilities to the `to_fix` key of the known flaws JSON file.
   - If the new vulnerability is considered a false positive, we add it to the `false_positives` list of the dictionary in its respective `known_flaws` JSON file.
   - If the new vulnerability is a real vulnerability, we solve the problem reported and remove the flaw from the known flaws file.
 
-The test also updates the known_flaws files automatically. If we have a look to a known_flaws file, we will see that each flaw dictionary contains information like the line number or range. This information si the one updated by the test. The test also removes flaws from the known_flaws file if they don't appear in the Bandit output. 
+The test also updates the known_flaws files automatically. If we have a look to a known_flaws file, we will see that each flaw dictionary contains information like the line number or range. This information si the one updated by the test. The test also removes flaws from the known_flaws file if they don't appear in the Bandit output.
```
Comment (Selutario) [inline tests/scans/code_analysis/README.md:41]:
```suggestion
The test also updates the known_flaws files automatically. If we have a look at a known_flaws file, we will see that each flaw dictionary contains information like the line number or range. This information is the one updated by the test. The test also removes flaws from the known_flaws file if they don't appear in the Bandit output.
```

**Review by Selutario — CHANGES_REQUESTED** [review]

**Review by Selutario — COMMENTED** [review]
Also, could you change the base branch to 4.5.2?

**Comment by nico-stefani** [comment]
I close this and we continue in #4439

**Pull request closed without merging**